rclpy/rclpy/node.py

```diff
@@ -1019,11 +1051,11 @@ def set_parameters_callback(
         """
         Register a set parameters callback.
 
-        Calling this function with override any previously registered callback.
+        Calling this function will add a callback in self._parameter_callback list.
 
         :param callback: The function that is called whenever parameters are set for the node.
         """
-        self._parameters_callback = callback
+        self.add_on_set_parameters_callback(callback)
```

Comment:
I feel like this should instead do

```
self._parameters_callback = [callback]
```
(in other words, should replace an existing list)

Reply:
@allenh1 sir but I think this will override the existing list of callbacks,instead of adding it to the list.

Reply:
> this will override the existing list of callbacks,instead of adding it to the list.

That's the point I was trying to make -- I think that should be the behavior

Reply:
@allenh1 sir but callbacks are to be added to the list instead of overriding them..

Reply:
> @allenh1 sir but callbacks are to be added to the list instead of overriding them..

I would expect that facility from a function called `add_on_set_parameters_callback` but not from one called `set_parameters_callback` since that name implies it replaces, not adds.

Reply:
@allenh1 so should I implement this..new test cases are to be made again for testing then.

Reply:
Completely up to @dirk-thomas. Tests need to be implemented for the new behavior, imho, but I think current tests cover this behavior.

Reply:
@allenh1 okay sir I will wait for @dirk-thomas

Reply:
@dirk-thomas sir is this okay to be merged?

Reply:
@suab321321 this probably still needs basic test cases to be added.

Reply:
As pointed out before (https://github.com/ros2/rclpy/pull/457#discussion_r345342727) changing the semantic of the existing API isn't desired. Please keep the original semantic.

Existing usage of this API might call it multiple times expecting the callback to be overwritten and don't use any `remove` API since it didn't exist until now.